src/utils/sshUtils.ts

```diff
@@ -13,14 +13,18 @@ import { cpUtils } from "./cpUtils";
 
 export const sshFsPath: string = join(os.homedir(), '.ssh');
 
-export async function getSshKey(vmName: string): Promise<string> {
+export async function getSshKey(vmName: string, passphrase?: string): Promise<string> {
     const sshKeyName: string = `azure_${vmName}_rsa`;
     const sshKeyPath: string = join(sshFsPath, sshKeyName);
 
+    // if passphrase is undefined, it'll cause a problem with generating the ssh key, so make it empty string
+    // tslint:disable-next-line: strict-boolean-expressions
+    passphrase = passphrase || '';
+
     if (!await fse.pathExists(`${sshKeyPath}.pub`)) {
         // if the SSH key doesn't exist, create it
-        const sshKeygenCmd: string = `ssh-keygen -t rsa -b 2048 -f ${sshKeyPath} -N ""`;
-        ext.outputChannel.appendLog(localize('generatingKey', `Generating public/private rsa key pair with command "${0}".`, sshKeygenCmd));
+        const sshKeygenCmd: string = `ssh-keygen -t rsa -b 4096 -f ${sshKeyPath} -N "${passphrase}"`;
```

Comment:
We should leverage `callWithMaskHandling` now that passphrase is involved

Reply:
Do you think it'd make more sense to just copy/paste that into this repo?  `callWithMaskHandling` is relatively simple code and exists in the `appservice` package which is a dependency not used by VMs.

Reply:
Personally I'd put it in the ui package. It's directly related to `callWithTelemetryAndErrorHandling` which is already in the ui package. I agree it's pretty simple, but it's the kind of code you want to be extra careful with, so I think sharing is worth it.